app/town_metadata.rb

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+require 'active_record'
+require 'active_support/core_ext/hash'
+require 'yaml'
+require 'nokogiri'
+require 'rack'
+require 'erb'
+
+module TownMetadata
+  class API
+    def connect_to_town_database
+      template = ERB.new File.new("config/settings.yml").read
+      @settings = YAML.load template.result(binding)
+
+      ActiveRecord::Base.establish_connection(
+        adapter:  'postgresql',
+        host: @settings['database']['host'],
+        port: @settings['database']['port'],
+        database: @settings['database']['town']['database'],
+        username: @settings['database']['username'],
+        password: @settings['database']['password'],
+        schema_search_path: @settings['database']['town']['schema']['metadata']
+      )
+    end
+
+    def metadata_for(tables, columns)
+      if tables
+        conditions = tables.map{|table| "LOWER(name) LIKE LOWER('#{table}')"}.join(' OR ')
+        where_clause = "WHERE #{conditions}"
+      else
+        where_clause = ''
+      end
+
+      sql = <<~SQL
+      SELECT
+        #{columns}
+      FROM
+        gdb_items
+      #{where_clause};
+      SQL
+      ActiveRecord::Base.connection.execute(sql)
+    end
+
+    def response(request)
+        connect_to_town_database
```

Comment:
Why is the body of this method doubly indented?

Reply:
Copy paste fail.